### PR TITLE
Changed SelectionButton from a StatelessWidget to a StatefulWidget

### PR DESCRIPTION
**Pull request by snat-s**
_Description of what this PR is changing or adding, and why:_
Changed the SelectionButton from a StatelessWidget to a StatefulWidget because the example could crash from [use_build_context_synchronously](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html) 

_Issues fixed by this PR (if any):_
Fixed a possible crash from the example.

## Presubmit checklist
- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by sfshaza2** [comment]
@domesticmouse, can you weigh in? 
@snat-s, aside from Brett's review, when you change a code excerpt, you need to refresh the code excerpts by running "./tool/refresh-code-excerpts.sh" from the top of the repo. That is what is breaking the build.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Need to refresh code excerpts.

**Comment by Piinks** [comment]
Self-pinging to my inbox to follow-up. @snat-s let me know if running the above command does not work. Thanks!

**Comment by snat-s** [comment]
For sure! I just don't know how to change the version of the diffutils. I'm currently on an M1 if anyone could help
Screen Shot 2022-07-01 at 16 44 37

**Comment by domesticmouse** [comment]
> For sure! I just don't know how to change the version of the diffutils. I'm currently on an M1 if anyone could help
> 
> Screen Shot 2022-07-01 at 16 44 37
> 
> 

Have you installed home brew? The brew version of `diffutils` is good IIRC.

**Comment by domesticmouse** [comment]
I created a GitHub codespace based on this PR and updating the code snippets reversed out the changes of this PR.

Here are the required changes to make this PR happy:

```diff
diff --git a/examples/cookbook/navigation/returning_data/lib/main.dart b/examples/cookbook/navigation/returning_data/lib/main.dart
index d7c54128..1a3b6373 100644
--- a/examples/cookbook/navigation/returning_data/lib/main.dart
+++ b/examples/cookbook/navigation/returning_data/lib/main.dart
@@ -25,9 +25,14 @@ class HomeScreen extends StatelessWidget {
   }
 }
 
-class SelectionButton extends StatelessWidget {
+class SelectionButton extends StatefulWidget {
   const SelectionButton({super.key});
 
+  @override
+  State<SelectionButton> createState() => _SelectionButtonState();
+}
+
+class _SelectionButtonState extends State<SelectionButton> {
   @override
   Widget build(BuildContext context) {
     return ElevatedButton(
@@ -49,6 +54,10 @@ class SelectionButton extends StatelessWidget {
       MaterialPageRoute(builder: (context) => const SelectionScreen()),
     );
 
+    // When a BuildContext is used from a StatefulWidget, the mounted property
+    // must be checked after an asynchronous gap.
+    if (!mounted) return;
+
     // After the Selection Screen returns a result, hide any previous snackbars
     // and show the new result.
     ScaffoldMessenger.of(context)
diff --git a/examples/cookbook/navigation/returning_data/lib/main_step2.dart b/examples/cookbook/navigation/returning_data/lib/main_step2.dart
index 439e4da5..d428eb29 100644
--- a/examples/cookbook/navigation/returning_data/lib/main_step2.dart
+++ b/examples/cookbook/navigation/returning_data/lib/main_step2.dart
@@ -30,9 +30,14 @@ class HomeScreen extends StatelessWidget {
 // #enddocregion HomeScreen
 
 // #docregion SelectionButton
-class SelectionButton extends StatelessWidget {
+class SelectionButton extends StatefulWidget {
   const SelectionButton({super.key});
 
+  @override
+  State<SelectionButton> createState() => _SelectionButtonState();
+}
+
+class _SelectionButtonState extends State<SelectionButton> {
   @override
   Widget build(BuildContext context) {
     return ElevatedButton(
diff --git a/examples/cookbook/navigation/returning_data/pubspec.yaml b/examples/cookbook/navigation/returning_data/pubspec.yaml
index 9df27d44..0f38fc24 100644
--- a/examples/cookbook/navigation/returning_data/pubspec.yaml
+++ b/examples/cookbook/navigation/returning_data/pubspec.yaml
@@ -9,7 +9,7 @@ dependencies:
     sdk: flutter
 
 dev_dependencies:
-  flutter_lints: ^1.0.3
+  flutter_lints: ^2.0.0
 
 flutter:
   uses-material-design: true
diff --git a/src/cookbook/navigation/returning-data.md b/src/cookbook/navigation/returning-data.md
index 5648448d..198cd86f 100644
--- a/src/cookbook/navigation/returning-data.md
+++ b/src/cookbook/navigation/returning-data.md
@@ -196,10 +196,11 @@ Future<void> _navigateAndDisplaySelection(BuildContext context) async {
     context,
     MaterialPageRoute(builder: (context) => const SelectionScreen()),
   );
+
   // When a BuildContext is used from a StatefulWidget, the mounted property
   // must be checked after an asynchronous gap.
-  if(!mounted) return;
-  
+  if (!mounted) return;
+
   // After the Selection Screen returns a result, hide any previous snackbars
   // and show the new result.
   ScaffoldMessenger.of(context)

```

File: examples/cookbook/navigation/returning_data/lib/main.dart [hunk 1]
```diff
@@ -54,6 +54,10 @@ class _SelectionButtonState extends State<SelectionButton> {
       MaterialPageRoute(builder: (context) => const SelectionScreen()),
     );
 
+
```
Comment (snat-s) [inline examples/cookbook/navigation/returning_data/lib/main.dart:57]:
```suggestion
```

**Comment by domesticmouse** [comment]
The snippets in need of changing are the following:

```diff
diff --git a/src/cookbook/navigation/returning-data.md b/src/cookbook/navigation/returning-data.md
index dbc3a82d..198cd86f 100644
--- a/src/cookbook/navigation/returning-data.md
+++ b/src/cookbook/navigation/returning-data.md
@@ -196,7 +196,7 @@ Future<void> _navigateAndDisplaySelection(BuildContext context) async {
     context,
     MaterialPageRoute(builder: (context) => const SelectionScreen()),
   );
-  
+
   // When a BuildContext is used from a StatefulWidget, the mounted property
   // must be checked after an asynchronous gap.
   if (!mounted) return;
@@ -268,7 +268,6 @@ class _SelectionButtonState extends State<SelectionButton> {
       MaterialPageRoute(builder: (context) => const SelectionScreen()),
     );
 
-
     // When a BuildContext is used from a StatefulWidget, the mounted property
     // must be checked after an asynchronous gap.
     if (!mounted) return;
```

The first involves deleting trailing white space, the second involves deleting a duplicate blank line.

File: src/cookbook/navigation/returning-data.md [hunk 2]
```diff
@@ -261,6 +268,9 @@ class _SelectionButtonState extends State<SelectionButton> {
       MaterialPageRoute(builder: (context) => const SelectionScreen()),
     );
 
+
```
Comment (snat-s) [inline src/cookbook/navigation/returning-data.md:271]:
```suggestion
```

Comment (snat-s) [inline src/cookbook/navigation/returning-data.md:271]:
```suggestion
```

Comment (snat-s) [inline src/cookbook/navigation/returning-data.md:271]:
```suggestion
```

**Comment by sfshaza2** [comment]
@khanhnwin, would you mind terribly to just pull in this PR, apply the excerpt refresh, and update? I'd like to land it.

**Comment by domesticmouse** [comment]
That should hopefully fix the CI

**Comment by Piinks** [comment]
Looks like CI passed! Can someone merge before it becomes outdated again? 😃

**Pull request merged**